packages/ckeditor5-image/src/image/replaceimagesourcecommand.ts

```diff
@@ -47,6 +47,13 @@ export default class ReplaceImageSourceCommand extends Command {
 			writer.setAttribute( 'src', options.source, image );
 			writer.removeAttribute( 'srcset', image );
 			writer.removeAttribute( 'sizes', image );
+
+			/**
+			 * In case responsive images some attributes should be cleaned up.
+			 * Check: https://github.com/ckeditor/ckeditor5/issues/15093
+			 */
+			writer.removeAttribute( 'ckboxImageId', image );
+			writer.removeAttribute( 'sources', image );
```

Comment:
Shouldn't the CKBox command listen to `ReplaceImageSourceCommand#execute()`? I do not think mixing logic between two different plugins is a good idea. The CKBox integration adds attributes, and it should take care of removing them. It would require more code, but I think it's worth it.

Reply:
The question: does it have to be in the same block or not? If yes, that would complicate it much more.

Reply:
Fair point.